test/solver/gmres_kernels.cpp

```diff
@@ -294,18 +339,27 @@ TEST_F(Gmres, GmresApplyOneRHSIsEquivalentToRef)
 
 TEST_F(Gmres, GmresApplyMultipleRHSIsEquivalentToRef)
 {
-    int m = 123;
-    int n = 5;
-    auto ref_solver = ref_gmres_factory->generate(mtx);
-    auto exec_solver = exec_gmres_factory->generate(d_mtx);
-    auto b = gen_mtx(m, n);
-    auto x = gen_mtx(m, n);
-    auto d_b = gko::clone(exec, b);
-    auto d_x = gko::clone(exec, x);
+    using gko::solver::gmres::orthog_method;
+    auto base_params = gko::clone(ref, ref_gmres_factory)->get_parameters();
 
-    ref_solver->apply(b, x);
-    exec_solver->apply(d_b, d_x);
-
-    GKO_ASSERT_MTX_NEAR(d_b, b, 0);
-    GKO_ASSERT_MTX_NEAR(d_x, x, r<value_type>::value * 1e3);
+    for (auto orthog :
+         {orthog_method::mgs, orthog_method::cgs, orthog_method::cgs2}) {
+        SCOPED_TRACE(orthog);
+        int m = 123;
+        int n = 5;
+        auto ref_solver =
+            base_params.with_orthog_method(orthog).on(ref)->generate(mtx);
+        auto exec_solver =
+            base_params.with_orthog_method(orthog).on(exec)->generate(d_mtx);
+        auto b = gen_mtx(m, n);
+        auto x = gen_mtx(m, n);
+        auto d_b = gko::clone(exec, b);
+        auto d_x = gko::clone(exec, x);
+
+        ref_solver->apply(b, x);
+        exec_solver->apply(d_b, d_x);
+
+        GKO_ASSERT_MTX_NEAR(d_b, b, 0);
```

Comment:
not needed
```suggestion
```

Reply:
I'm not sure I understand. For this test, I put the existing test "guts" inside a loop over each of the different orthogonalization methods -- if it was included before, why not now?

Reply:
IMO the test should also test what might be changed by the function. `apply` is guaranteed to not modify `b/d_b`, as they are maked `const`. So there is no need to test that they are same.

Reply:
I agree it seems odd, but it's also done in the `cgs`, `gcr`, and `idr` tests (in test/solver), so I'm hesitant to change in this PR.

Reply:
Yeah, I would say it is not really necessary in those either. So, I would say you can remove them in all those tests as well.

Reply:
Should that be in this PR, though? For all the tests that have it, it seems to go all the way back to the original addition of the tests -- except GMRES, where it was added in 2022? (https://github.com/ginkgo-project/ginkgo/commit/963b1a3b152c797da61b76cbbc16e89d8f3ad3c9#diff-79b1284e17363ce1e01a13254f43119d1a20c7973d1477efe43c79c3938711fc). I think we should keep this PR limited to GMRES as far as code that it touches.

Reply:
Then, I guess we can tackle it in a future PR, if necessary.